### PR TITLE
Feature|nav bar

### PR DESCRIPTION
**Pull request by antek-ka**
10/08:
- project setup
- setup global sass imports in vite.config
- create sass variables partial file

11/08:
- install & setup vite-svg-loader
- reconfigure project from sass to scss

12/08:
- create main stylesheet w/ style reset
- create global mixins & variables
- create BaseButton component
- create responsive Navbar w/ subcomponents
-create draft Homepage component & render Navbar component

Navbar-1440px:
navbar-1440

Navbar-1024px:
navbar-1024

Navbar-375px_closed:
navbar-375_closed

Navbar-375px_open:
navbar-375_open

### CODE REVIEW

File: src/components/global/nav-bar/nav-bar.vue [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            MonteCinemaIcon,
+            ToggleNavOpenIcon,
+            ToggleNavClosedIcon,
+            NavLinks,
+            NavButtonContainer,
+        },
+        props: {
+            userType: {
+                type: String,
+                default: 'public', //Others: client, employee
```
Comment (saviorsoul) [inline src/components/global/nav-bar/nav-bar.vue]: You can add prop `validator` here with check for proper values.

Prop validations are mostly liked for thing like this - they inform developer what values he can use quickly : )

File: src/components/global/nav-bar/nav-bar.vue [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                class="nav-toggle-icon" 
+                v-if="isScreenSmall" 
+                @click="toggleNavOpen"
+            >
+                <ToggleNavOpenIcon v-if="!isNavOpen" />
+                <ToggleNavClosedIcon v-if="isNavOpen" />
+            </div>
+        </div>
+        <NavLinks v-if="isScreenSmall && isNavOpen || !isScreenSmall" />
+        <NavButtonContainer v-if="isScreenSmall && isNavOpen || !isScreenSmall" />
```
Comment (saviorsoul) [inline src/components/global/nav-bar/nav-bar.vue]: Let's follow the DRY principle and create method with long enough name to describe what it does and reuse it : )

File: src/main.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { createApp } from 'vue'
 import App from './App.vue'
 
-import './assets/main.css'
+import '../src/assets/styles/main.scss'
```
Comment (saviorsoul) [inline src/main.js:4]: Use `@` path to `assets` to make use of what is defined in `vite.config.js` ->  `resolve.alias`

File: src/components/global/nav-bar/nav-button-container.vue [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<script>
+    import { defineComponent } from 'vue';
+
+    import BaseButton from '../base-button.vue';
```
Comment (saviorsoul) [inline src/components/global/nav-bar/nav-button-container.vue]: Use `@` path to point to `base-button.vue`

File structure might change and at least you won't have to deal with imports inside files that changes its own path : )

File: src/assets/styles/main.scss [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+header, 
+nav, 
+section, 
+main {
+    display: block;
+}
+
+body {
+    line-height: 1;
+}
```
Comment (saviorsoul) [inline src/assets/styles/main.scss:77]: `line-height: 1` is useful for texts inside buttons or in other custom components. Browsers has different initial values of `line-height` and it's worth to normalise it but the `1` on `body` element is not a great choice in terms of readability ; )

File: src/components/global/base-button.vue [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+<style lang="scss">
+    .base-button {
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        border: 2px solid transparent;
+        border-radius: $button-radius;
+        font-weight: $fw-500;
+        line-height: 100%;
```
Comment (saviorsoul) [inline src/components/global/base-button.vue:54]: `line-height: 100%` === `line-height: 1`

worth to stick with one notation in the whole app

File: src/components/global/nav-bar/nav-bar.vue [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+        data() {
+            return {
+                isScreenSmall: false,
+                isNavOpen: false,
+            }
+        },
+        methods: {
+            watchScreenSmall() {
+                this.isScreenSmall = window.innerWidth < 1024;
+            },
```
Comment (saviorsoul) [inline src/components/global/nav-bar/nav-bar.vue]: Maybe we can avoid checking the window size using JavaScript and define CSS value which hides content on a given breakpoints using `display: none`?

CSS should be responsible for visual part as often as possible : )

File: src/components/global/nav-bar/nav-links.vue [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+import { defineComponent } from 'vue';
+
+export default defineComponent({
+   name: 'NavLinks',
+   data() {
+    return {
+        urls: [
+            {id:1, displayName: 'Movies', url: '#'},
+            {id:2, displayName: 'Screenings', url: '#'},
+            {id:3, displayName: 'Contact us', url: '#'},
```
Comment (saviorsoul) [inline src/components/global/nav-bar/nav-links.vue]: spaces after `id:`

File: src/pages/home-page.vue [hunk 9]
```diff
@@ -0,0 +1,26 @@
+<script>
```
Comment (saviorsoul) [inline src/pages/home-page.vue:1]: `<script setup>`

File: src/assets/styles/main.scss (hunk 5 above)
Comment (antek-ka) [reply src/assets/styles/main.scss:77]: Just want to clear this up - should I change the unit in which the property is declared, or simply remove this rule?

File: src/components/global/nav-bar/nav-bar.vue (hunk 1 above)
Comment (antek-ka) [reply src/components/global/nav-bar/nav-bar.vue]: Thanks for the tip!
Haven't reached this section in the docs yet.. :)

File: src/components/global/nav-bar/nav-bar.vue (hunk 7 above)
Comment (antek-ka) [reply src/components/global/nav-bar/nav-bar.vue]: Ok, done!

File: src/components/global/nav-bar/nav-bar.vue (hunk 2 above)
Comment (antek-ka) [reply src/components/global/nav-bar/nav-bar.vue]: That will not be necessary as I  found another - more CSS'y - solution.

File: src/components/global/nav-bar/nav-button-container.vue (hunk 4 above)
Comment (antek-ka) [reply src/components/global/nav-bar/nav-button-container.vue]: Good point, I updated the paths throughout the project.

**Pull request closed without merging**